---
title: "Figure1"
author: "Mike Vermeulen"
date: '2022-03-02'
output: html_document
---

```{r setup, include=FALSE}
library(Seurat)
library(magrittr)
library(data.table)
library(Signac)
library(Biobase)
library(GEOquery)
library(ggpubr)
library(Repitools)
library(tidyverse)
library(dittoSeq)
library(harmony)
library(viridis)
library(tibble)
library(reshape2)
library(scDblFinder)
library(SingleR)
library(celldex)
suppressPackageStartupMessages(library(EnsDb.Hsapiens.v86))
edb <- EnsDb.Hsapiens.v86

setwd("H:/LOY/scLOY/scLOY/")
source("E:/LOY/Example/PACKAGE/call_LOY.R")
source("E:/LOY/scLOY/scripts/loy_functions.R")
source('H:/LOY/scLOY/scLOY/R/load.R')
source('H:/LOY/scLOY/scLOY/data_raw/build_data_files.R')
source("E:/LOY/scLOY/scripts/annotation_functions.R")
```

```{r load_data}

readRDS("e:/LOY/scLOY/data/10X_MULTIOME_LYMPH/seurat_MV.RDS") -> o


```

## Multi-modal LOY analysis (Fig 1)

Expression-based classification of LOY agrees with ATAC-seq in the same nuclei.

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

### Fig 1A

UMAP plot showing clustering of nuclei using expression only.

```{r analysis_fig1a}


######## UMAPS #######
dittoSeq::dittoDimPlot(object = o, var = "cell_ident_MV", size = 0.5,
                       do.label = T, labels.size = 3) %>% ggpar(title = "", legend = "none") + Seurat::NoAxes() -> p


p

ggsave(plot = p, filename = "A:/Dropbox/LOY_microglia_paper/GR_RESUB/PLOTS/Fig1/CELL_TYPE_UMAP.pdf",
       dpi = "retina", width = 4, height = 4, units = "in")


```

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

### Fig 1B

```{r analysis_fig1b}


factor(o@meta.data$cell_ident_MV,
       levels = rev(c("B tumor","B","T","Cycling T","Myeloid","pDC","Stromal"))) -> o@meta.data$cell_ident_MV

# 
DotPlot(object = o, dot.scale = 13, features = rev(c("CEMIP","CLEC4C","CD14","CD163","MKI67",
                                 "CD3E","TRAC","CD8A","CD4",
                                 "BANK1","MS4A1",
                                 "PAX5","GPM6A")), group.by = "cell_ident_MV", cols = c("white","#3C53A4")) %>%
                                    ggpar(x.text.angle = 45, font.xtickslab = c(15,"bold"), 
                                          font.ytickslab = c(17,"bold"),
                                          ylab = "", xlab ="Genes") -> p 

p

#ggsave(plot = p, filename = "A:/Dropbox/LOY_microglia_paper/GR_RESUB/PLOTS/Fig1/DOTPLOT.pdf",
#       dpi = "retina", width = 7.74, height = 5, units = "in")

```

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

### Fig 1C

```{r analysis_fig1c}

dittoSeq::dittoDimPlot(object = o, var = "LOY", do.label = F, size = 0.4,
                       color.panel = c("red","grey"),
                       order = c("decreasing")) %>% ggpar(title = "", legend = "none") + NoAxes() -> p

#ggsave(plot = p, filename = "A:/Dropbox/LOY_microglia_paper/GR_RESUB/PLOTS/Fig1/LOY_UMAP.pdf",
#       dpi = "retina", width = 3, height = 3, units = "in")

```

### Fig 1D

```{r analysis_fig1d}

dittoSeq::dittoBarPlot(object = o, group.by = "cell_ident_MV", var = "LOY",
                       color.panel = c("grey","red"), var.labels.reorder = c(2,1)) -> p
p %>% ggpar(title = "", xlab = "Cell type", legend = "none", ylab = "Proportion of cells",
            font.xtickslab = c(12,"bold"), font.ytickslab = c(12), font.y = c(12), font.x = c(12)) -> p

#ggsave(plot = p, filename = "A:/Dropbox/LOY_microglia_paper/GR_RESUB/PLOTS/Fig1/LOY_PROP_BAR.pdf",
#       dpi = "retina", width = 4, height = 4, units = "in")

```

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

### Fig 1G

```{r analysis_fig1g}


```

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

### Fig 1F

```{r}

# inset boxplot (left)

## UMI cutoff 1000
ggboxplot(o@meta.data[o@meta.data$nCount_RNA > 1000,], x = "LOY", y = "Norm_Access") + stat_compare_means()
o@meta.data[o@meta.data$nCount_RNA > 1000,] %>% 
  dplyr::group_by(LOY) %>% 
  summarise(median_normalized_gene_activity_MSY = median(Norm_Access), n = dplyr::n())

## UMI cutoff 3000
ggboxplot(o@meta.data[o@meta.data$nCount_RNA > 3000,], x = "LOY", y = "Norm_Access") + stat_compare_means()
o@meta.data[o@meta.data$nCount_RNA > 3000,] %>% 
  dplyr::group_by(LOY) %>% 
  summarise(median_normalized_gene_activity_MSY = median(Norm_Access), n = dplyr::n())


# inset boxplot (right)
## UMI cutoff 1000
ggboxplot(o@meta.data[o@meta.data$nCount_RNA > 1000 & o@meta.data$cell_ident_MV=="B tumor",],
          x = "LOY", y = "Norm_Access", fill = "LOY")  %>%
  ggpar(palette = c("red","grey"), legend = "none",
        xlab = "", ylab = "Gene activity (chrY)",
        font.xtickslab = c(12), x.text.angle = 45) -> out
out + stat_compare_means()

## UMI cutoff 3000
ggboxplot(o@meta.data[o@meta.data$nCount_RNA > 3000 & o@meta.data$cell_ident_MV=="B tumor",],
          x = "LOY", y = "Norm_Access", fill = "LOY")  %>%
  ggpar(palette = c("red","grey"), legend = "none",
        xlab = "", ylab = "Gene activity (chrY)",
        font.xtickslab = c(12), x.text.angle = 45) -> out
out + stat_compare_means()


# Density plot
# Density plots UMI cutoff 1000
ggdensity(data = o@meta.data[o@meta.data$nCount_RNA > 1000,], x = "Norm_Access", ylab = "Density",
          xlab = "Gene activity (chrY)",
          fill = "LOY", palette = c("red","grey")) + coord_cartesian(xlim = c(0,0.2)) -> p 
          ggpar(p,legend.title = "LOY classification",
                title = "UMI cutoff 1000 (all cell-types)")  -> p1

ggdensity(data = o@meta.data[o@meta.data$nCount_RNA > 1000 & o@meta.data$cell_ident_MV=="B tumor",],
          x = "Norm_Access", ylab = "Density", 
          xlab = "Gene activity (chrY)",
          fill = "LOY", palette = c("red","grey")) + coord_cartesian(xlim = c(0,0.2)) -> p; 
          ggpar(p, legend.title = "LOY classification", title = "UMI cutoff 1000 (B tumor only)")  -> p2

#ggarrange(plotlist = list(p1,p2), common.legend = T, legend = "bottom") 


```

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

### SUPP FIG 2

Supplementary Fig. 2 \| Y chromosome ATAC gene activity is exclusively repressed in LOY nuclei. Boxplots comparing LOY and non-LOY/normal nuclei chromatin accessibility-based gene activity across all chromosomes (n=9,380). Gene activity scores were calculated using genes with \> 0.2 normalized expression across all nuclei from the chromosome of interest. Boxes represent the 25th percentile, median, and 75th percentile. The whiskers extend to the furthest value that is no more than 1.5 times the inter-quartile range. LOY was classified using expression.

```{r S2}

DefaultAssay(o) <- "RNA"
o[["ATAC_gene_activity"]]@data -> data
data %>% as.matrix() -> data
edb <- EnsDb.Hsapiens.v86

## only use genes that are expressed
AverageExpression(o, group.by = "orig.ident") -> t
t$RNA[t$RNA > 1.090402e-02,] %>% names() -> elig


lapply(c(1:22,"X","Y"), function(i){
  message(i)
  edb.x <- addFilter(edb, SeqNameFilter(i))
  genes(edb.x) -> tmp
  Repitools::annoGR2DF(tmp) -> tmp
  AverageExpression(subset(o, nCount_RNA > 3000), features = tmp$gene_name) -> dat

  dat$ATAC_gene_activity[rownames(dat$ATAC_gene_activity) %in% elig,] %>% 
    as.data.frame() -> k

  k$chr <- i
  return(k)

}) -> output
data.table::rbindlist(output) -> output

reshape2::melt(output, id = "chr" ) -> output

ggpubr::ggboxplot(output, x = "chr", y = "value", 
                  fill = "variable", outlier.shape = NA,
                  palette = c("red","grey")) + ylim(0,0.75) -> p
                  p %>% ggpar(xlab = "", ylab = "Gene activity (ATAC-seq)", 
                  legend = "none", font.xtickslab = c(8,"bold")) -> p

#ggsave(plot = p, filename = "A:/Dropbox/LOY_microglia_paper/GR_RESUB/PLOTS/Fig1/LOY_PROP_BAR.pdf",
#       dpi = "retina", width = 4, height = 4, units = "in")

```



### SUPP FIG 3
```{r S3}

#### S3A
Idents(o) <- "LOY"
DefaultAssay(o) <- "ATAC"
p1 <- CoveragePlot(
  object = subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000), tile = F,
  region = "UTY",
  features = "UTY", window = 2000,
  expression.assay = "RNA",
  extend.upstream = 1000,
  extend.downstream = 500
)

#ggsave(plot = p, filename = "A:/Dropbox/LOY_microglia_paper/GR_RESUB/PLOTS/Fig1/LOY_PROP_BAR.pdf",
#       dpi = "retina", width = 4, height = 4, units = "in")



#### S3B
Idents(o) <- "LOY"
DefaultAssay(o) <- "ATAC"
p2 <- CoveragePlot(
  object = subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000), tile = F,
  region = "ZFY",
  features = "ZFY", window = 2000,
  expression.assay = "RNA",
  extend.upstream = 100000,
  extend.downstream = 500
)

#ggsave(plot = p, filename = "A:/Dropbox/LOY_microglia_paper/GR_RESUB/PLOTS/Fig1/LOY_PROP_BAR.pdf",
#       dpi = "retina", width = 4, height = 4, units = "in")




```
