---
title: "Figure1"
author: "Mike Vermeulen"
date: '2022-03-02'
output: html_document
---

```{r setup, include=FALSE}
library(Seurat)
library(magrittr)
library(data.table)
library(Signac)
library(Biobase)
library(GEOquery)
library(ggpubr)
library(tidyverse)
library(scibetR)
library(harmony)
library(viridis)
library(tibble)
library(reshape2)
library(scDblFinder)
library(SingleR)
library(celldex)
suppressPackageStartupMessages(library(EnsDb.Hsapiens.v86))
edb <- EnsDb.Hsapiens.v86

setwd("H:/LOY/scLOY/scLOY/")
source("E:/LOY/Example/PACKAGE/call_LOY.R")
source("E:/LOY/scLOY/scripts/loy_functions.R")
source('H:/LOY/scLOY/scLOY/R/load.R')
source('H:/LOY/scLOY/scLOY/data_raw/build_data_files.R')
source("E:/LOY/scLOY/scripts/annotation_functions.R")
```

## R Markdown

```{r input_data}

Read10X_h5("e:/LOY/scLOY/data/10X_MULTIOME_LYMPH/lymph_node_lymphoma_14k_filtered_feature_bc_matrix.h5") -> h5

```

## Preprocess

```{r preprocess, echo=FALSE}
h5$`Gene Expression` -> expr

CreateSeuratObject(counts = expr, min.cells = 10, min.features = 200 ) -> o
o[["percent.mt"]] <- PercentageFeatureSet(o, pattern = "^MT-")
NormalizeData(o) -> o

o@meta.data$donor_organism.sex <- "male"
LOY_genes_to_exclude_expr(o) -> remove
call_LOY(s_obj = o, genes_to_exclude = remove) -> o

subset(o, nCount_RNA > 1000 & nFeature_RNA > 800) -> o

h5$Peaks -> peaks

grange.counts <- StringToGRanges(rownames(peaks), sep = c(":", "-"))
grange.use <- seqnames(grange.counts) %in% standardChromosomes(grange.counts)
peaks <- peaks[as.vector(grange.use), ]
annotations <- GetGRangesFromEnsDb(ensdb = EnsDb.Hsapiens.v86)
seqlevelsStyle(annotations) <- 'UCSC'
genome(annotations) <- "hg38"

peaks[grepl(x = rownames(peaks), pattern = "chrY"),] %>% colSums() -> chrY

dplyr::left_join( x = tibble::rownames_to_column(o@meta.data,"CB"),
                  y = tibble::rownames_to_column(as.data.frame(chrY),"CB") ,
                  by = "CB") %>% tibble::column_to_rownames("CB") -> o@meta.data

o@meta.data %>% group_by(LOY) %>% summarise(median = median(chrY))

chrom_assay <- Signac::CreateChromatinAssay(
  counts = peaks,
  sep = c(":", "-"),
  genome = 'hg38',
  fragments = 'e:/LOY/scLOY/data/10X_MULTIOME_LYMPH/lymph_node_lymphoma_14k_atac_fragments.tsv.gz',
  annotation = annotations
)

colnames(chrom_assay)[colnames(chrom_assay) %in% colnames(o)] -> keep

subset(chrom_assay, cells = keep ) -> chrom_assay
subset(o, cells = keep) -> o

o[["ATAC"]] <- chrom_assay

dplyr::mutate(o@meta.data, chrY_ratio_ATAC = chrY/nCount_ATAC) -> o@meta.data

ifelse(o@meta.data$LOY == "LOY" & o@meta.data$chrY == 0, "LOY", "NORMAL") -> o@meta.data$multimodal_LOY

FindMarkers(subset(o, nCount_RNA > 1500), ident.1 = "LOY", ident.2 = "NORMAL", group.by = "multimodal_LOY")


DefaultAssay(o) <- "ATAC"
o <- NucleosomeSignal(o)
o <- TSSEnrichment(o)


VlnPlot(
  object = o,
  features = c("nCount_RNA", "nCount_ATAC", "TSS.enrichment", "nucleosome_signal"),
  ncol = 4,
  pt.size = 0
)

o <- subset(
  x = o,
  subset = nCount_ATAC < 100000 &
    nCount_RNA < 20000 &
    nCount_ATAC > 1000 &
    nCount_RNA > 1000 &
    nucleosome_signal < 1.5 &
    TSS.enrichment > 1
)







DefaultAssay(o) <- "RNA"

o@meta.data$cell_ident_MV <- o@meta.data$RNA_snn_res.0.1
levels(o@meta.data$cell_ident_MV)[levels(o@meta.data$cell_ident_MV)%in%c(2)] <- "Myeloid"
levels(o@meta.data$cell_ident_MV)[levels(o@meta.data$cell_ident_MV)%in%c(1)] <- "B tumor"
levels(o@meta.data$cell_ident_MV)[levels(o@meta.data$cell_ident_MV)%in%c(6)] <- "pDC"
levels(o@meta.data$cell_ident_MV)[levels(o@meta.data$cell_ident_MV)%in%c(5)] <- "Stromal"
levels(o@meta.data$cell_ident_MV)[levels(o@meta.data$cell_ident_MV)%in%c(3)] <- "Cycling T"
levels(o@meta.data$cell_ident_MV)[levels(o@meta.data$cell_ident_MV)%in%c(0)] <- "T"
levels(o@meta.data$cell_ident_MV)[levels(o@meta.data$cell_ident_MV)%in%c(4)] <- "B"
DimPlot(o, group.by = "cell_ident_MV", label = T)

```

```{r load_processed}
# Have this processed file hosted for download if possible
# The above pre-processing can be included in a different file
readRDS("e:/LOY/scLOY/data/10X_MULTIOME_LYMPH/seurat_MV.RDS") -> o


## ADD gene activity scores

o[["ATAC_gene_activity"]]@data -> data
data %>% as.matrix() -> data
data[rownames(data) %in% expressed_Y_genes(o),] -> Y_counts
Y_counts %>% colMeans() %>% as.data.frame() -> Y
names(Y) <- "Norm_Access"
left_join(x = tibble::rownames_to_column(o@meta.data,"CB"), y = tibble::rownames_to_column(Y,"CB"), by = "CB" ) %>%
  tibble::column_to_rownames("CB") -> o@meta.data
```


## Analysis (Fig 1)

### Fig 1A

```{r analysis_fig1a}

######## UMAPS #######
dittoSeq::dittoDimPlot(object = o, var = "cell_ident_MV", size = 0.5,
                       do.label = T, labels.size = 2) %>% ggpar(title = "") + Seurat::NoAxes()

```


### Fig 1B
```{r analysis_fig1b}


factor(o@meta.data$cell_ident_MV,
       levels = rev(c("B tumor","B","T","Cycling T","Myeloid","pDC","Stromal"))) -> o@meta.data$cell_ident_MV

# 3C53A4
DotPlot(object = o, dot.scale = 13, features = rev(c("CEMIP","CLEC4C","CD14","CD163","MKI67",
                                 "CD3E","TRAC","CD8A","CD4",
                                 "BANK1","MS4A1",
                                 "PAX5","GPM6A")), group.by = "cell_ident_MV", cols = c("white","#3C53A4")) %>%
                                    ggpar(x.text.angle = 45, font.xtickslab = c(12,"bold"), 
                                          ylab = "", xlab ="Genes") 

```


### Fig 1C

```{r analysis_fig1c}

dittoSeq::dittoDimPlot(object = o, var = "LOY", do.label = F, size = 0.4,
                       color.panel = c("red","grey"),
                       order = c("decreasing")) %>% ggpar(title = "") + NoAxes() 

```


### Fig 1D

```{r analysis_fig1d}

dittoSeq::dittoBarPlot(object = o, group.by = "cell_ident_MV", var = "LOY",
                       color.panel = c("grey","red"), var.labels.reorder = c(2,1)) -> p
p %>% ggpar(title = "", xlab = "Cell type", legend = "bottom") 

```


### Fig 1G

```{r analysis_fig1g}


```



### Fig 1F
```{r}

# inset boxplot (left)

## UMI cutoff 1000
ggboxplot(o@meta.data[o@meta.data$nCount_RNA > 1000,], x = "LOY", y = "Norm_Access") + stat_compare_means()
o@meta.data[o@meta.data$nCount_RNA > 1000,] %>% 
  dplyr::group_by(LOY) %>% 
  summarise(median_normalized_gene_activity_MSY = median(Norm_Access), n = dplyr::n())

## UMI cutoff 3000
ggboxplot(o@meta.data[o@meta.data$nCount_RNA > 3000,], x = "LOY", y = "Norm_Access") + stat_compare_means()
o@meta.data[o@meta.data$nCount_RNA > 3000,] %>% 
  dplyr::group_by(LOY) %>% 
  summarise(median_normalized_gene_activity_MSY = median(Norm_Access), n = dplyr::n())


# inset boxplot (right)
## UMI cutoff 1000
ggboxplot(o@meta.data[o@meta.data$nCount_RNA > 1000 & o@meta.data$cell_ident_MV=="B tumor",],
          x = "LOY", y = "Norm_Access", fill = "LOY")  %>%
  ggpar(palette = c("red","grey"), legend = "none",
        xlab = "", ylab = "Gene activity (chrY)",
        font.xtickslab = c(12), x.text.angle = 45) -> out
out + stat_compare_means()

## UMI cutoff 3000
ggboxplot(o@meta.data[o@meta.data$nCount_RNA > 3000 & o@meta.data$cell_ident_MV=="B tumor",],
          x = "LOY", y = "Norm_Access", fill = "LOY")  %>%
  ggpar(palette = c("red","grey"), legend = "none",
        xlab = "", ylab = "Gene activity (chrY)",
        font.xtickslab = c(12), x.text.angle = 45) -> out
out + stat_compare_means()


# Density plot
# Density plots UMI cutoff 1000
ggdensity(data = o@meta.data[o@meta.data$nCount_RNA > 1000,], x = "Norm_Access", ylab = "Density",
          xlab = "Gene activity (chrY)",
          fill = "LOY", palette = c("red","grey")) + coord_cartesian(xlim = c(0,0.2)) -> p 
          ggpar(p,legend.title = "LOY classification",
                title = "UMI cutoff 1000 (all cell-types)")  -> p1

ggdensity(data = o@meta.data[o@meta.data$nCount_RNA > 1000 & o@meta.data$cell_ident_MV=="B tumor",],
          x = "Norm_Access", ylab = "Density", 
          xlab = "Gene activity (chrY)",
          fill = "LOY", palette = c("red","grey")) + coord_cartesian(xlim = c(0,0.2)) -> p; 
          ggpar(p, legend.title = "LOY classification", title = "UMI cutoff 1000 (B tumor only)")  -> p2

#ggarrange(plotlist = list(p1,p2), common.legend = T, legend = "bottom") 


```

