---
title: "Figure2-microglia-loss-of-y"
author: "Mike Vermeulen"
date: '2022-03-04'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(Seurat)
library(magrittr)
library(data.table)
library(ggpubr)
library(Repitools)
library(tidyverse)
library(dittoSeq)
library(harmony)
library(viridis)
library(tibble)
library(reshape2)
library(devtools)
library(readr)
suppressPackageStartupMessages(library(EnsDb.Hsapiens.v86))
edb <- EnsDb.Hsapiens.v86


devtools::source_url("https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/scripts/LOYtools/functions.R")

# filters LOY tables
filter_function <- function(min_cells=50, score = -1.25,l , 
                            type = c("Oligodendrocyte","Microglia","Astrocyte","Neuron","OPC","Pericyte")){
## cleans up 
  
l[l$TOTAL_CELLS > min_cells,] -> l
l[l$cell_type %in% type,] -> dat
dat[!grepl(dat$file, pattern = "MD5893"),] -> dat
dat[!grepl(dat$tissue, pattern = "Lung"),] -> dat
dat[which(dat$sample=="Microglia_MO_MCI3"),]$neuro_degen_diagnosis <- "MCI"
dat[dat$neuro_degen_diagnosis %in% c("Alzheimer's disease"),]$neuro_degen_diagnosis <- "AD"
dat[which(dat$neuro_degen_diagnosis=="Diffuse Lewy Body disease, limbic or transitional type"),]$neuro_degen_diagnosis <- "PD"
dat[dat$neuro_degen_diagnosis %in% c("Non-symptomatic","Normal"),]$neuro_degen_diagnosis <- "Control"


dat$sum_Y_exp -> x
scale(x) -> dat$sum_Y_exp_scaled

dat$expr_Y -> x
scale(x) -> dat$expr_Y_scaled
 
dplyr::mutate(.data = dat, scaled_Y_metric  = (sum_Y_exp_scaled*1.5 + expr_Y_scaled*1) / 2) -> dat

dat[dat$sum_Y_exp_scaled >= score,] -> dat
return(dat)
  
}

# function to pull p value from lm model
lmp <- function (modelobject) {
  if (class(modelobject) != "lm") stop("Not an object of class 'lm' ")
  f <- summary(modelobject)$fstatistic
  p <- pf(f[1],f[2],f[3],lower.tail=F)
  attributes(p) <- NULL
  return(p)
}

# l[l$CLEAN=="Disease state",]$CLEAN <- l[l$CLEAN=="Disease state",]$neuro_degen_diagnosis

### SAVE
save <- F


```


\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

### Fig 2a


```{r pre_filter_stats }


### TOTAL CELLS IN STUDY (PRE FILTER)

readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_1500_1000.txt") %>%
   as.data.table() -> l

filter_function(min_cells = 0, score = -2, l = l) -> dat
paste0(dat$sample, "_", dat$cohort) -> dat$ID
gsub(pattern = "_cleaned_filtered", x = dat$cohort, replacement = "") -> dat$cohort

dat %>% dplyr::group_by(cohort) %>% dplyr::summarise(`Total cells`=sum(TOTAL_CELLS)) -> out

if(save == T){
data.table::fwrite(x = out, file = "h:/LOY/scLOY/scLOY_NOV18/plots/MARCH_12/MANUSCRIPT/TABLES/UPDATED_MALE_CELLS_1500.csv")}

## how many samples w/ LOY
dat %>% dplyr::group_by(ID) %>% dplyr::summarise(mean = mean(LOY_prop)) -> out.
out.[out.$mean > 0,] %>% nrow()

## number of cells/nuclei
out$`Total cells` %>% sum() -> cells

## number of samples
# dat %>% group_by(cohort, sample) %>% summarize(n = dplyr::n()) %>% nrow()
dat %>% group_by(cohort, sample) %>% summarize(n = dplyr::n(), age = mean(donor_organism.age)) -> tmp
tmp$sample %>% length() -> s

paste0("Before filtering our dataset was comprised of ", cells, " cells/nuclei from ", s, " donors")
paste0("This includes all CNS cells/nuclei with >1500 UMI and >1000 genes")



```
\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

```{r SuppX_min_Yexpr_threshold}


readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(min_cells = 100, score = -10, l = l,
                type = c("Astrocyte","Neuron","OPC","Microglia","Oligodendrocyte","Pericyte","CAM","Monocyte","Fibroblast")) -> dat
paste0(dat$sample, "_", dat$cohort) -> dat$ID
gsub(pattern = "_cleaned_filtered", x = dat$cohort, replacement = "") -> dat$cohort

# sum_Y_exp
dat$LOY_percent <- dat$LOY_prop * 100 %>% as.numeric()
gghistogram(dat, x = "sum_Y_exp", facet.by = "cell_type")
ggscatter(dat, x = "sum_Y_exp_scaled", y = "LOY_percent", facet.by = "cell_type", size = 0.4) -> p 
  ggpar(p, xlab = "ChrY exp sparsity score (chrY-wt populations)", ylab = "Loss of Y (%)") + geom_vline(xintercept = -1.25) -> p
  
  
dat$LOY_percent <- dat$LOY_prop * 100 %>% as.numeric()
ggscatter(dat, x = "sum_Y_exp_scaled", y = "LOY_percent", size = 0.9, color = "cell_type") -> p 
  ggpar(p, xlab = "ChrY exp sparsity score (chrY-wt populations)", legend.title = "Cell type", palette = dittoColors(), legend = "bottom",
        ylab = "Loss of Y (%)") + geom_vline(xintercept = -1.25) -> p
  p + geom_smooth(se = T) -> p
  
if(save == T){
ggsave(plot = p, file = "a:/Dropbox/LOY_microglia_paper/GR_RESUB/MANUSCRIPT/PLOTS/Supp_QC/chrY_sparsity_threshold.pdf",
       dpi = "retina",units = "in", width = 7, height = 5)}
  

  ggscatter(dat, x = "sum_Y_exp_scaled", y = "LOY_prop", size = 0.6, add = "loess", facet.by = "cell_type") -> p 
  ggpar(p, xlab = "ChrY exp sparsity score") 
  
  ggscatter(dat, x = "expr_Y_scaled", y = "LOY_prop", size = 0.6, add = "loess", color = "cell_type") -> p 
  ggpar(p, xlab = "ChrY expression score") 


## threshold number of cells/nuclei
gghistogram(dat, x = "TOTAL_CELLS", facet.by = "cell_type")
ggscatter(dat, x = "TOTAL_CELLS", y = "LOY_prop", facet.by = "cell_type", size = 0.4) + geom_vline(xintercept = 75) -> p 
 ggpar(p, xlab = "Total cells/nuclei available")
 



  
  ## bin the scaled_Y_metric and color by PAR difference between LOY and NORMAL
```  
  

```{r SuppX_panel_thresholds}
  
readr::read_csv(
"https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
as.data.table() -> l

filter_function(min_cells = 0, score = -1.25, l = l,type = c("Astrocyte","Neuron","OPC","Microglia",
                                                             "Oligodendrocyte","Pericyte","CAM","Fibroblast")) -> dat


dat$LOY_percent <- dat$LOY_prop * 100 %>% as.numeric()
ggscatter(data = dat, y = "LOY_percent", x = "TOTAL_CELLS", color = "cell_type", size = 0.8, 
          palette = dittoSeq::dittoColors()[c(7,8,9,5,4,1,6,2,3)] ) + 
          geom_vline(xintercept = 100) -> p 
          ggpar(p, legend = "bottom", legend.title = "Cell type", ylim = c(0,100),
                ylab = "Loss of Y (%)",
                xlab = "Donor/cell-type | Sample size (n)") + coord_cartesian(xlim = c(0,2500)) -> p1
          
  if(save == T){
ggsave(plot = p1, file = "a:/Dropbox/LOY_microglia_paper/GR_RESUB/MANUSCRIPT/PLOTS/Supp_QC/min_cells_threshold_scatter.pdf",
       dpi = "retina",units = "in", width = 7, height = 5)}

#####

filter_function(min_cells = 100, score = -10, l = l,
                type = c("Astrocyte","Neuron","OPC","Microglia","Oligodendrocyte","Pericyte","CAM","Fibroblast")) -> dat
paste0(dat$sample, "_", dat$cohort) -> dat$ID
gsub(pattern = "_cleaned_filtered", x = dat$cohort, replacement = "") -> dat$cohort

# sum_Y_exp
dat$LOY_percent <- dat$LOY_prop * 100 %>% as.numeric()
gghistogram(dat, x = "sum_Y_exp", facet.by = "cell_type")
ggscatter(dat, x = "sum_Y_exp_scaled", y = "LOY_percent", facet.by = "cell_type", size = 0.4) -> p 
  ggpar(p, xlab = "ChrY exp sparsity score (chrY-wt populations)", ylab = "Loss of Y (%)") + geom_vline(xintercept = -1.25) -> p
  
  
dat$LOY_percent <- dat$LOY_prop * 100 %>% as.numeric()
ggscatter(dat, x = "sum_Y_exp_scaled", y = "LOY_percent", size = 0.9, color = "cell_type") -> p 
  ggpar(p, xlab = "ChrY exp sparsity score (chrY-wt populations)", legend.title = "Cell type", palette = dittoColors(), legend = "bottom",
        ylab = "Loss of Y (%)") + geom_vline(xintercept = -1.25) -> p
  p + geom_smooth(se = T) -> p2
  
if(save == T){
ggsave(plot = p2, file = "a:/Dropbox/LOY_microglia_paper/GR_RESUB/MANUSCRIPT/PLOTS/Supp_QC/chrY_sparsity_threshold.pdf",
       dpi = "retina",units = "in", width = 7, height = 5)}
        

  ggarrange(plotlist = list(p1,p2),common.legend = T, legend = "bottom")  -> p3
  
  if(save == T){
ggsave(plot = p3, file = "a:/Dropbox/LOY_microglia_paper/GR_RESUB/MANUSCRIPT/PLOTS/Supp_QC/chrY_sparsity_Min_cells.pdf",
       dpi = "retina",units = "in", width = 9, height = 5)}
  
```

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

```{r post_filter_stats}
### TOTAL CELLS IN STUDY (POST FILTER)

readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(min_cells = 75, sum_Y = 250, l = l) -> dat
paste0(dat$sample, "_", dat$cohort) -> dat$ID
gsub(pattern = "_cleaned_filtered", x = dat$cohort, replacement = "") -> dat$cohort

dat %>% dplyr::group_by(cohort) %>% dplyr::summarise(`Total cells`=sum(TOTAL_CELLS)) -> out

paste0("We applied strict QC filtering to our dataset to isolate high-information cells/nuclei ",
"allowing for more reliable loss of chrY calling")

paste0("Cells/nuclei were included if they had >3000 UMI counts and >1000 genes")
paste0("Cell type clusters from individual samples were included if total cells/nuclei >75")
paste0("We also tested chrY expression in non-LOY nuclei in each cell type population in each sample")
paste0("Cell type populations in specific samples were removed ",
       "if MSY genes were not significantly expressed in WT (non-LOY) male cells.")

if(save == T){
data.table::fwrite(x = out, file = "h:/LOY/scLOY/scLOY_NOV18/plots/MARCH_12/MANUSCRIPT/TABLES/UPDATED_MALE_CELLS_1500.csv")}

## how many samples w/ LOY
#dat %>% dplyr::group_by(ID) %>% dplyr::summarise(mean = mean(LOY_prop)) -> out.
#out.[out.$mean > 0,] %>% nrow()

## number of cells/nuclei
out$`Total cells` %>% sum() -> tot

## number of samples
# dat %>% group_by(cohort, sample) %>% summarize(n = dplyr::n()) %>% nrow()
dat %>% group_by(cohort, sample) %>% summarize(n = dplyr::n(), age = mean(donor_organism.age)) -> tmp
tmp$sample %>% length() -> s 

## age data available
is.na(tmp$age) %>% table() %>% .[1] -> age_data

paste0("After filtering our dataset was comprised of ", tot, " cells/nuclei from ", s, " donors")

paste0("Of these donors ", age_data, " had available age data")
paste0("Mean age was ", round(mean(tmp$age, na.rm = T),2), " range: ", 
       round(min(tmp$age, na.rm = T)) ,"-",
       round(max(tmp$age, na.rm = T)))



###### SAMPLE SIZE | NEURO DIAG
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(min_cells = 50,l = l) -> dat
paste0(dat$sample, "_", dat$cohort) -> dat$ID
dat %>% dplyr::distinct(ID, .keep_all = T ) %>% dplyr::group_by(neuro_degen_diagnosis) %>% dplyr::summarise(count = dplyr::n())


# net LOY prop
dat %>% group_by(cell_type) %>% summarise(LOY = sum(LOY_cells), NORMAL = sum(NORMAL_cells)) %>%
  dplyr::mutate(total = LOY + NORMAL, prop = LOY / total) -> out


paste0("Of all included cells/nuclei ", 100 * round((sum(out$LOY) / sum(out$total)), 4), "% were classified as LOY")

if(save == T){
data.table::fwrite(out, file = "h:/LOY/scLOY/scLOY_NOV18/plots/MARCH_12/MANUSCRIPT/TABLES/summarized_cell_type_LOY.csv")}


# LOY % by sample 
paste0(dat$sample,"_",dat$cohort) -> dat$ID
dat %>% group_by(cohort, cell_type) %>% summarise(LOY = sum(LOY_cells), NORMAL = sum(NORMAL_cells)) %>%
  dplyr::mutate(total = LOY + NORMAL, prop = LOY / total) -> out


### NUMBER OF SAMPLES WITH GREATER THAN 10, 15, 20, 25% LOY ######
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l
filter_function(min_cells = 75, score = -2, l = l) -> dat
paste0(dat$sample, "_", dat$cohort) -> dat$ID

dat[dat$LOY_prop > 0.10,]$cell_type %>% table()
dat[dat$LOY_prop > 0.15,]$cell_type %>% table()
dat[dat$LOY_prop > 0.20,]$cell_type %>% table()
dat[dat$LOY_prop > 0.25,]$cell_type %>% table()
```


\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

```{r diseases_included}






```


\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_



```{r LOY_frequency_cell_type_barplot}

### LOY FREQUENCY BY CELL_TYPE BARPLOT ######
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(l = l, min_cells = 75) -> dat

factor(dat$cell_type, levels = c("Microglia","Pericyte", "OPC","Oligodendrocyte","Astrocyte","Neuron"),
       labels = c("Microglia","Pericyte", "OPC","OL","Astrocyte","Neuron")) -> dat$cell_type
dat %>% dplyr::group_by(cell_type) %>% dplyr::summarise(LOY = sum(LOY_cells), NORMAL = sum(NORMAL_cells)) %>%
  dplyr::mutate(LOY_prop = LOY / (LOY + NORMAL), TOTAL = LOY + NORMAL) %>%
  ggbarplot(x = "cell_type", y = "LOY_prop", palette = dittoSeq::dittoColors()[c(5,8,7,1,6,9)], fill = "cell_type") %>%
    ggpar(x.text.angle = 45, font.xtickslab = c(10,"bold"), ylab = "LOY proportion", xlab = "") %>% ggpar(legend = "none") -> p

plots <- "h:/LOY/scLOY/scLOY_NOV18/plots/MARCH_12/MANUSCRIPT/MAIN/Fig2"
ggsave(plot = p,
       filename = paste0(plots,"/LOY_proportion_by_cell_type_sum.pdf"),
       dpi = "retina", units = "in", height = 3, width = 2)
```

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

```{r main_plots}
##### MAIN PLOTS ##############
plots <- "h:/LOY/scLOY/scLOY_NOV18/plots/MARCH_12/MANUSCRIPT/MAIN/Fig2"

### NEURO + CELLTYPE |  LOY
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

#
#
#
#
# Clean 
#
#
#
#

#l[l$TOTAL_CELLS > 75 & l$sum_Y_exp >= 250, ] -> l
#l[l$cell_type%in%c("Oligodendrocyte","Microglia","Astrocyte","Neuron","OPC","Pericyte"),] -> dat
#dat[!grepl(dat$file, pattern = "MD5893"),] -> dat
#dat[!grepl(dat$cohort, pattern = "GSE131907_LUNG_NSCLC_cleaned_filtered"),] -> dat
#data.table::fwrite(dat, file = "h:/LOY/scLOY/scLOY_NOV18/plots/MARCH_12/MANUSCRIPT/TABLES/sample_cell_type_LOY_data.csv")


filter_function(l = l, min_cells = 75) -> dat

factor(dat$cell_type, levels = c("Microglia","Pericyte", "OPC","Oligodendrocyte","Astrocyte","Neuron"),
       labels = c("Microglia","Pericyte", "OPC","OL","Astrocyte","Neuron")) -> dat$cell_type

factor(dat$CLEAN, levels = c("Non-degenerative","AD","HD","PD","Other dementias","ALS","MS")) -> dat$CLEAN
ggboxplot(dat[dat$cell_type %in% c("Microglia","Astrocyte","Neuron","OPC","OL"),],
          x = "CLEAN", font.xtickslab = c(8), size = 0.3,
          xlab = "", ylab = "Loss of Y proportion", fill = "CLEAN", palette = "aaas", add = "jitter", add.params = list(size = 0.9),
          y = "LOY_prop") %>% ggpar(x.text.angle = 45, ylim = c(0,0.7), legend = "none") -> tmp
tmp + facet_grid(~ cell_type) -> tmp

if(save == T){
ggsave(plot = tmp,
       filename = paste0(plots,"/NEURO_LOY_BRAIN_CELL_TYPES_3000_1000_no_jitter.pdf"), dpi = "retina", units = "cm", height = 9, width = 13)}



ggstripchart(dat[dat$cell_type %in% c("Microglia","Astrocyte","Neuron","OPC","OL"),], jitter = 0.1,
             x = "CLEAN", font.xtickslab = c(8), size = 0.85, add = "median_iqr",
             add.params = list(color = "CLEAN"),
             xlab = "", ylab = "Loss of Y proportion", fill = "neuro_degen_diagnosis", palette = "aaas",
             y = "LOY_prop") %>% ggpar(x.text.angle = 45, ylim = c(0,0.6), legend = "none") -> tmp
tmp + facet_grid(~ cell_type) -> tmp

plots <- "h:/LOY/scLOY/scLOY_NOV18/plots/MARCH_12/MANUSCRIPT/MAIN/Fig2"


### CELL TYPE | LOY ##########
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(l = l, min_cells = 100) -> dat
factor(dat$cell_type, levels = c("Microglia","Pericyte", "Oligodendrocyte","OPC","Astrocyte","Neuron"),
       labels = c("Microglia","Pericyte", "OL","OPC","Astrocyte","Neuron")) -> dat$cell_type

dat %>% group_by(sample,cell_type,cohort) %>% summarize(mean_LOY = mean(LOY_prop) , prop = LOY_prop) -> tmp

fun_mean <- function(x){
  return(data.frame(y=mean(x),label=mean(x,na.rm=T)))}


ggboxplot(dat[dat$cell_type %in% c("Microglia","Astrocyte","Neuron","OPC","OL"),], add = "mean",
          x = "cell_type", font.xtickslab = c(6), size = 0.2, outlier.size = 0.8, add.params = list(color = "black", shape = "17"),
          xlab = "", ylab = "LOY proportion", fill = "cell_type", palette = dittoSeq::dittoColors()[c(5,1,6,7,8)],
          y = "LOY_prop") %>%
          ggpar(x.text.angle = 45, legend = "none",
                font.xtickslab = c(10,"bold"), font.y = c(10,"bold") ) + coord_cartesian(ylim = c(0,0.6)) -> tmp
if(save == T){ 
ggsave(plot = tmp,
       filename = paste0(plots,"/LOY_BRAIN_CELL_TYPES_3000_1000_no_jitter_.pdf"),
       dpi = "retina", units = "in", height = 6, width = 3)}


### IF THE OUTLIERS ARE REMOVED (>20% LOY) HOW DOES IT AFFECT THE AD-LOY ASSOCIATION? and CELL TYPE LOY rates.
ggboxplot(dat[dat$cell_type %in% c("Microglia","Astrocyte","Neuron","OPC","OL") &
                !(dat$file %in% c("A163_17_samples_stat_GSE160936_3000_1000_Microglia.csv",
                                    "A096_14_samples_stat_GSE160936_3000_1000_Microglia.csv",
                                    "Microglia_MO_MCI3_samples_stat_syn12514624_3000_1000_Microglia.csv",
                                    "3881_samples_stat_GSE152058_3000_1000_OPC.csv")),], add = "mean",
          x = "cell_type", font.xtickslab = c(6), size = 0.2, outlier.size = 0.8,
          add.params = list(color = "black", shape = "17",size = 0.5),
          xlab = "", ylab = "LOY proportion", fill = "cell_type", palette = dittoSeq::dittoColors()[c(5,1,6,7,8)],
          y = "LOY_prop") %>%
  ggpar(x.text.angle = 45, legend = "none",
        font.xtickslab = c(10,"bold"), font.y = c(10,"bold") ) + coord_cartesian(ylim = c(0,0.3)) -> tmp

if(save == T){ 
ggsave(plot = tmp,
       filename = paste0(plots,"/LOY_BRAIN_CELL_TYPES_3000_1000_no_jitter_no_outliers.pdf"),
       dpi = "retina", units = "in", height = 6, width = 3)}

# Mean of sample LOY
dat[dat$cell_type %in% c("Microglia","Astrocyte","Neuron","OPC","OL") &
      !(dat$file %in% c("A163_17_samples_stat_GSE160936_3000_1000_Microglia.csv",
                        "A096_14_samples_stat_GSE160936_3000_1000_Microglia.csv",
                        "Microglia_MO_MCI3_samples_stat_syn12514624_3000_1000_Microglia.csv",
                        "3881_samples_stat_GSE152058_3000_1000_OPC.csv")),] %>%
      dplyr::group_by(cell_type) %>%
      dplyr::summarise(mean = mean(LOY_prop), median = median(LOY_prop))

# Total LOY
dat[dat$cell_type %in% c("Microglia","Astrocyte","Neuron","OPC","OL") &
      !(dat$file %in% c("A163_17_samples_stat_GSE160936_3000_1000_Microglia.csv",
                        "A096_14_samples_stat_GSE160936_3000_1000_Microglia.csv",
                        "Microglia_MO_MCI3_samples_stat_syn12514624_3000_1000_Microglia.csv",
                        "3881_samples_stat_GSE152058_3000_1000_OPC.csv")),] %>%
  dplyr::group_by(cell_type) %>%
  summarize(total = sum(TOTAL_CELLS), LOY = sum(LOY_cells)) %>% dplyr::mutate(LOY / total)


# Mean of sample LOY
dat[dat$cell_type %in% c("Microglia","Astrocyte","Neuron","OPC","OL"),] %>%
  dplyr::group_by(cell_type) %>%
  dplyr::summarise(mean = mean(LOY_prop), median = median(LOY_prop))

# Total LOY
dat[dat$cell_type %in% c("Microglia","Astrocyte","Neuron","OPC","OL"),] %>%
  group_by(cell_type) %>%
  summarize(total = sum(TOTAL_CELLS), LOY = sum(LOY_cells)) %>% dplyr::mutate(LOY / total)




### NEURO + CELLTYPE |  LOY #########
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

l[l$TOTAL_CELLS > 75 & l$sum_Y_exp >= 250, ] -> l
l[l$cell_type%in%c("Oligodendrocyte","Microglia","Astrocyte","Neuron","OPC"),] -> dat
dat[dat$neuro_degen_diagnosis=="Normal",]$neuro_degen_diagnosis <- "Control"
dat[!grepl(dat$file, pattern = "MD5893"),] -> dat
dat[!grepl(dat$cohort, pattern = "GSE131907_LUNG_NSCLC_cleaned_filtered"),] -> dat

factor(x = dat$cell_type, levels = c("Microglia","Astrocyte","Neuron","Oligodendrocyte","OPC"),
       labels = c("Microglia","Astrocyte","Neuron","OL","OPC")) -> dat$cell_type
dat[which(dat$sample=="Microglia_MO_MCI3"),]$neuro_degen_diagnosis <- "AD"

dat[dat$cell_type=="Microglia" & dat$neuro_degen_diagnosis %in% c("Control","AD"),]


ggboxplot(dat[dat$cell_type %in% c("Microglia","Astrocyte","Neuron","OPC","OL"),],
          x = "neuro_degen_diagnosis", font.xtickslab = c(6), size = 0.2, outlier.size = 0.2, add = "jitter",
          xlab = "", ylab = "Loss of Y proportion", fill = "neuro_degen_diagnosis", palette = "aaas",
          y = "LOY_prop") %>% ggpar(x.text.angle = 45, ylim = c(0,0.6), legend = "none") -> tmp
tmp + facet_grid(~ cell_type) -> tmp

if(save == T){ 
ggsave(plot = tmp,
       filename = paste0(plots,"/NEURO_LOY_BRAIN_CELL_TYPES_3000_1000_no_jitter.tiff"), dpi = "retina",
       units = "cm", height = 9, width = 13)}
```

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_


```{r age_LOY_plots}

##### AGE + CELL TYPE | LOY #######

readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(l = l, min_cells = 100, score = -1) -> dat
dat[!is.na(dat$donor_organism.age),] -> tmp


factor(x = tmp$cell_type, levels = c("Microglia","Astrocyte","Neuron","Oligodendrocyte","OPC"),
       labels = c("Microglia","Astrocyte","Neuron","OL","OPC")) -> tmp$cell_type

tmp[!is.na(tmp$donor_organism.age),] -> tmp
tmp[tmp$cell_type %in% c("Microglia","Astrocyte","Neuron","OL","OPC") &
      !(tmp$donor_organism.age < 0),] -> tmp

cut(x = tmp$donor_organism.age, breaks = c(0,30,60,80,120),
    labels = c("0-29","30-59","60-79","80-100")) -> tmp$quantile_age



tmp[tmp$cell_type=="",] %>%
  ggscatter(x = "donor_organism.age",y = "LOY_prop") + stat_cor()



my_comparisons <- list( c("71-80","81-100"), c("61-70","81-100"))
tmp[tmp$cell_type=="Microglia",] %>% dplyr::group_by(cohort,sample) %>% 
  dplyr::summarise(LOY = sum(LOY_cells), total = sum(TOTAL_CELLS), age = quantile_age) %>%
  dplyr::mutate(LOY_prop = LOY / total) %>%
  ggboxplot(x = "age", fill = "age", size = 0.2, add = "jitter",
            xlab = "", ylab = "Loss of Y proportion", outlier.size = 0.5,
            y = "LOY_prop") %>% ggpar(x.text.angle = 45, ylim = c(0,0.6), legend = "none", font.xtickslab = c(8)) +
            stat_compare_means(comparisons = my_comparisons ) -> e

ggboxplot(tmp, x = "quantile_age", fill = "quantile_age", size = 0.2,
          xlab = "", ylab = "Loss of Y proportion", outlier.size = 0.5, add = "jitter",
          y = "LOY_prop") %>% ggpar(x.text.angle = 45, ylim = c(0,0.6), legend = "none", font.xtickslab = c(8)) -> e
e + facet_grid(~ cell_type) + scale_fill_brewer(palette = "Greens") -> e

if(save == T){ 
ggsave(plot = e, filename = paste0(plots,"/AGE_BRAIN_CELL_TYPES_no_jitter_3000_1000.tiff"), dpi = "retina", units = "in",
       height = 4.76, width = 6)}

tmp[!is.na(tmp$donor_organism.age),] -> tmp
tmp[tmp$cell_type %in% c("Microglia","Astrocyte","Neuron","OL","OPC") &
      !(tmp$quantile_age %in% c("Prenatal")),] -> tmp
ggboxplot(tmp, x = "quantile_age", fill = "quantile_age", size = 0.2,
          xlab = "", ylab = "Loss of Y proportion", outlier.size = 0.5,
          y = "LOY_prop") %>% ggpar(x.text.angle = 45, ylim = c(0,0.6), legend = "none", font.xtickslab = c(8)) -> e
e + scale_fill_brewer(palette = "Greens") -> e
e + ggplot2::coord_cartesian(ylim = c(0,0.2)) -> e


if(save == T){ 
ggsave(plot = e, filename = paste0(plots,"/AGE_BRAIN_no_jitter_3000_1000.tiff"), dpi = "retina", units = "in",
       height = 4, width = 3) }

ggscatter(tmp,x = "donor_organism.age", y = "LOY_prop", add = "reg.line", size = 1, color = "cell_type", palette = "npg",
          add.params = list(color = "blue")) + stat_cor()


#### AGE SCATTER #############
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(l=l, min_cells = 100, score = -1.25, type = c("Microglia","Astrocyte","Neuron","Oligodendrocyte","OPC","Pericyte")) -> dat

cut(x = dat$donor_organism.age, breaks = c(-20,0,30,60,80,100),
    labels = c("Prenatal","0-29","30-59","60-79","80-100")) -> dat$quantile_age
dat[!is.na(dat$donor_organism.age),] -> tmp

factor(x = tmp$cell_type, levels = c("Microglia","Astrocyte","Neuron","Oligodendrocyte","OPC","Pericyte"),
       labels = c("Microglia","Astrocyte","Neuron","OL","OPC","Pericyte")) -> tmp$cell_type

tmp[!is.na(tmp$donor_organism.age),] -> tmp
tmp[tmp$cell_type %in% c("Microglia","Astrocyte","Neuron","OL","OPC"),] -> tmp

tmp %>%
  ggscatter(x = "donor_organism.age",y = "LOY_prop", color = "CLEAN",
            size = 1, add.params = list(color="black"),
            palette = get_palette(palette = "npg",6)) +
            geom_smooth(method = "lm", se = F) + stat_cor() + facet_grid(~cell_type) -> p
            ggpar(p, font.xtickslab = c(9,"bold"), x.text.angle = 45)

            
tmp[tmp$CLEAN %in% c("Non-degenerative"),] %>%
  ggscatter(x = "donor_organism.age",y = "LOY_prop", color = "CLEAN", size = 1, add.params = list(color="black"),
            palette = "npg") + geom_smooth(method = "lm", se = F) + stat_cor() + facet_grid(CLEAN~cell_type) -> p
            ggpar(p, font.xtickslab = c(9,"bold"), x.text.angle = 45)


### CELLTYPE |  LOY | NEURO BROAD #########
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(l =l , min_cells = 75, sum_Y = 250) -> dat
paste0(dat$sample, "_", dat$cohort) -> dat$ID
gsub(pattern = "_cleaned_filtered", x = dat$cohort, replacement = "") -> dat$cohort




factor(x = dat$cell_type, levels = c("Microglia","Astrocyte","Neuron","Oligodendrocyte","OPC"),
       labels = c("Microglia","Astrocyte","Neuron","OL","OPC")) -> dat$cell_type

dat[dat$cell_type %in% c("Microglia","Astrocyte","Neuron","OPC","OL"),] -> dat

ifelse(dat$neuro_degen_diagnosis %in% c("AD/MCI","HD","PD","ALS","LBD/FTLD"),
       "Neurodegenerative disorder", "Control") -> dat$neuro_degen_diagnosis_broad

factor(dat$CLEAN, levels = c("Non-degenerative","AD","LBD/FTLD","PD","HD","ALS","MS")) -> dat$CLEAN

ggboxplot(dat,
          x = "cell_type", font.xtickslab = c(6), size = 0.2, outlier.size= 0.6, add = "jitter",
          add.params = list(size = 0.4),
          xlab = "", ylab = "Loss of Y proportion", color = "CLEAN", palette = "aaas", fill = "CLEAN",
          y = "LOY_prop") %>% ggpar(x.text.angle = 45, ylim = c(0,0.6)) -> tmp


if(save == T){ 
ggsave(plot = tmp,
       filename = paste0(plots,"/NEURO_BROAD_LOY_BRAIN_CELL_TYPES_3000_1000_no_jitter_AD_MCI.tiff"), dpi = "retina",
       units = "in", height = 3.2, width = 2.8) }

# JUST IN MICROGLIA
ggboxplot(dat[dat$cell_type=="Microglia",],
          x = "CLEAN", font.xtickslab = c(9), size = 0.2, outlier.size= 0.6, add = "jitter", 
          add.params = list(color = "black", size = 0.4),
          xlab = "", ylab = "Loss of Y proportion", color = "CLEAN", palette = "aaas",
          y = "LOY_prop") %>% ggpar(x.text.angle = 45, ylim = c(0,0.6), legend = "none") + stat_compare_means() -> tmp

if(save == T){ 
ggsave(plot = tmp,
       filename = paste0(plots,"/MICROGLIA_LOY_BRAIN_CELL_TYPES_3000_1000_no_jitter_AD_MCI.pdf"), dpi = "retina",
       units = "in", height = 4, width = 4) }


factor(dat$neuro_degen_diagnosis_broad) -> dat$neuro_degen_diagnosis_broad
as.numeric(dat$donor_organism.age) -> dat$donor_organism.age
as.numeric(dat$LOY_prop) -> dat$LOY_prop
dat[dat$cell_type=="Microglia" & dat$neuro_degen_diagnosis %in% c("Control","AD/MCI"),] -> d

d[!is.na(d$donor_organism.age),] -> d

#d <- data.frame(LOY_prop = c(0.1,0.05,0.02,0.1,0.12,0.12,0.06,0.02,0.02,0.01,0.07,0.09,
#                             0.3,0.2,0.4,0.23,0.2,0.5,0.06,0.5,0.34,0.2,0.23,0.24),
#                neuro_degen_diagnosis_broad = c(rep("C",12),rep("N",12)),
#                donor_organism.age = c(50,50,52,56,62,42,45,56,62,67,23,56,
#                                       90,92,98,97,90,92,78,80,80,85,84,90))

factor(d$neuro_degen_diagnosis_broad) -> d$neuro_degen_diagnosis_broad
factor(d$neuro_degen_diagnosis) -> d$neuro_degen_diagnosis
d[d$sequencing_method2=="3'",]$sequencing_method2 <- "single-cell"

ggboxplot(d, x = "neuro_degen_diagnosis", y = "LOY_prop", add = "jitter", xlab = "Diagnosis", ylab = "LOY proportion") +
  stat_compare_means() -> p1

model <- lm(LOY_prop ~ neuro_degen_diagnosis + donor_organism.age,
            data = d)

ggboxplot(d[!(d$sample %in% c("Microglia_MO_MCI3","A163_17","A096_14")),], add= "jitter",
          xlab = "Diagnosis", ylab = "LOY proportion",
          x = "neuro_degen_diagnosis", y = "LOY_prop") + stat_compare_means() -> p2

model <- lm(LOY_prop ~ neuro_degen_diagnosis + donor_organism.age,
            data = d[!(d$sample %in% c("Microglia_MO_MCI3","A163_17","A096_14")),])

ggarrange(plotlist = list(p2,p1))



model <- lm(LOY_prop ~ donor_organism.age,
            data = d)

plot(model, add.smooth = FALSE, which = 1)
plot(model, which = 2)
plot(model, add.smooth = FALSE, which = 3)
anova(model)

model <- lm(LOY_prop ~ donor_organism.age,
            data = d)
summary(model)


ggboxplot(d, x = "neuro_degen_diagnosis", y = "LOY_prop", add = "jitter", fill = "neuro_degen_diagnosis") %>%
  ggpar(xlab = "", ylab = "LOY frequency", palette = "npg", font.tickslab = c(10,"bold"),
        font.y = c(12,"bold"), legend = "none")
nrow(d) # n = 41
table(d$neuro_degen_diagnosis)  # 15 AD / 26 Control
d %>% group_by(neuro_degen_diagnosis) %>% summarize(median_age = mean(donor_organism.age)) # 82.9 AD / 73.7 Control

#(ANCOVA: F=3.60,df=1,37,p=0.066) # AD diagnosis (microglia)
#(ANCOVA: F=1.02,df=1,37,p=0.317) # age (microglia)

wilcox.test(LOY_prop ~ neuro_degen_diagnosis, data = d, conf.int = TRUE)
#(Mann-Whitney U-test: U=273, n1=15, n2=26, p=0.0348)
# difference in location: 0.0189
# 95% conf = 0.002-0.0556

library(boot)
med.diff <- function(dat, i) {
  tmp <- dat[i,]
  median(tmp$weight[tmp$neuro_degen_diagnosis=="AD"]) -
    median(tmp$weight[tmp$neuro_degen_diagnosis=="Control"])
}
boot.out <- boot(data = d, statistic = med.diff, R = 1000)








dat %>% group_by(cell_type, neuro_degen_diagnosis_broad) %>%
  summarize(mean_LOY_prop = mean(LOY_prop), n = dplyr::n(), sd = sd(LOY_prop)) %>%
  reshape2::dcast(cell_type ~ neuro_degen_diagnosis_broad, value.var = c("mean_LOY_prop"))


dat[dat$cell_type=="Microglia" & neuro_degen_diagnosis_broad == "Neurodegenerative disorder",]$LOY_prop %>% mean(na.rm = T)
dat[dat$cell_type=="Microglia" & neuro_degen_diagnosis_broad == "Control",]$LOY_prop %>% mean(na.rm = T)







```

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

```{r fig2a}



##### FIG2A INSET #########################  
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(l, min_cells = 75, sum_Y = 250) -> dat
paste0(dat$sample, "_", dat$cohort) -> dat$ID
dat %>% dplyr::group_by(ID, neuro_degen_diagnosis) %>% dplyr::summarise(count = dplyr::n())

#data.table::fwrite(dat, file = "h:/LOY/scLOY/scLOY_NOV18/plots/MARCH_12/MANUSCRIPT/TABLES/sample_cell_type_LOY_data.csv")
dat %>% group_by(cell_type) %>% summarise(LOY = sum(LOY_cells), NORMAL = sum(NORMAL_cells)) %>%
  dplyr::mutate(total = LOY + NORMAL, prop = LOY / total, n = dplyr::n()) -> out

if(save == T){
data.table::fwrite(out, file = "h:/LOY/scLOY/scLOY_NOV18/plots/MARCH_12/MANUSCRIPT/TABLES/summarized_cell_type_neuro_LOY.csv")}


factor(out$cell_type, levels = c("Microglia","Pericyte","OPC","Oligodendrocyte","Astrocyte","Neuron"), 
       labels = c("Microglia","Pericyte","OPC","OL","Astrocyte","Neuron")) -> out$cell_type
t <- ggbarplot(data = out, x = "cell_type", y = "prop", fill = "cell_type", position = position_dodge2(),
          palette = dittoSeq::dittoColors()[c(5,9,8,7,1,6)]) %>% ggpar(ylab = "LOY proportion", xlab = "Cell type",
                                                                     x.text.angle = 45,
                                                                     legend = "bottom", font.legend = c(8,"bold"),
                                                                   legend.title = "Cell type", font.xtickslab = c(10,"bold"))


if(save == T){ 
ggsave(plot = t + Seurat::NoLegend(),
       filename = paste0(plots,"/Neuro_LOY_prop_totals_bar.pdf"), dpi = "retina", units = "in", height = 4, width = 3) }


t



##### Fig 2A ##################################

readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l
filter_function(l = l, min_cells = 75, sum_Y = 250) -> dat

factor(dat$cell_type, levels = c("Neuron","Oligodendrocyte","Astrocyte","Microglia","OPC","Pericyte"),
       labels = c("Neuron","OL","Astrocyte","Microglia","OPC","Pericyte")) -> dat$cell_type

factor(dat$neuro_degen_diagnosis,
       levels = rev(c("Control","AD/MCI","ALS","LBD/FTLD","HD","PD","Seizure disorder","Brain cancer"))) -> dat$neuro_degen_diagnosis


dat %>% group_by(cell_type, neuro_degen_diagnosis) %>% summarise(LOY = sum(LOY_cells), NORMAL = sum(NORMAL_cells)) %>%
  dplyr::mutate(total = LOY + NORMAL, prop = LOY / total) -> out

p <- ggbarplot(out, x = "cell_type", y= "total", 
               fill = "neuro_degen_diagnosis", palette = get_palette("npg",9)[c(9,2,5,7,4,3,1,6)], size = 0.2) %>%
  ggpar(xlab = "", ylab = "Total cells/nuclei", legend.title = "Diagnosis", legend = "bottom", font.xtickslab = c(10,"bold"))

plots <- "h:/LOY/scLOY/scLOY_NOV18/plots/MARCH_12/MANUSCRIPT/MAIN/Fig2"


if(save == T){ 
ggsave(plot = p,

       filename = paste0(plots,"/total_cells_cell_type_diagnosis_bar.pdf"),
       dpi = "retina", units = "in", height = 4.5, width = 6) }

p



### LOY PROPORTIONs

data.table::fread(paste0(output,"/LOY_prop_table_3000_1000.txt")) -> l
filter_function(l = l, min_cells = 75, sum_Y = 250) -> dat

dat %>% group_by(cell_type) %>% summarise(LOY = sum(LOY_cells), NORMAL = sum(NORMAL_cells)) %>%
  dplyr::mutate(total = LOY + NORMAL, prop = LOY / total) -> out

out

```

\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_

```{r fig2b}
### CELL TYPE | LOY ##########
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(l = l, min_cells = 75, sum_Y = 250) -> dat
factor(x = dat$cell_type, levels = c("Microglia","OPC","Oligodendrocyte","Astrocyte","Neuron"),
       labels = c("Microglia","OPC","OL","Astrocyte","Neuron")) -> dat$cell_type


paste0(dat$cohort, "_", dat$sample) -> dat$id

dat %>% group_by(sample,cell_type,cohort) %>% summarize(mean_LOY = mean(LOY_prop) , prop = LOY_prop) -> tmp

dat %>% group_by(cell_type) %>% summarize(counts = dplyr::n()) 

fun_mean <- function(x){
  return(data.frame(y=mean(x),label=mean(x,na.rm=T)))}



ggboxplot(dat[dat$cell_type %in% c("Microglia","Astrocyte","Neuron","OPC","OL"),], add = c("mean"), 
          x = "cell_type", font.xtickslab = c(6), size = 0.2, outlier.size = 0.8, add.params = list(color = "black", shape = "17"),
          xlab = "", ylab = "LOY proportion", fill = "cell_type", palette = dittoSeq::dittoColors()[c(5,8,7,1,6,9)],
          y = "LOY_prop") %>%
          ggpar(x.text.angle = 45, legend = "none",
                font.xtickslab = c(10,"bold"), font.y = c(10,"bold") ) + coord_cartesian(ylim = c(0,0.6)) -> tmp

if(save == T){ 
ggsave(plot = tmp,
       filename = paste0(plots,"/LOY_BRAIN_CELL_TYPES_3000_1000_no_jitter_.pdf"),
       dpi = "retina", units = "in", height = 4, width = 3)}



ggboxplot(dat[dat$cell_type %in% c("Microglia","Astrocyte","Neuron","OPC","OL"),], add = c("jitter","mean"), 
          x = "cell_type", font.xtickslab = c(6), size = 0.2, outlier.size = 0.8,
          xlab = "", ylab = "LOY proportion", fill = "cell_type", palette = dittoSeq::dittoColors()[c(5,8,7,1,6)],
          y = "LOY_prop") %>%
          ggpar(x.text.angle = 45, legend = "none",
                font.xtickslab = c(10,"bold"), font.y = c(10,"bold") ) + coord_cartesian(ylim = c(0,0.3)) -> tmp

ggboxplot(dat[dat$cell_type %in% c("Microglia","Neuron"),], add = c("jitter","mean"), 
          x = "cell_type", font.xtickslab = c(6), size = 0.2, outlier.size = 0.8,
          xlab = "", ylab = "LOY proportion", fill = "cell_type", palette = dittoSeq::dittoColors()[c(5,8,7,1,6)],
          y = "LOY_prop") %>%
          ggpar(x.text.angle = 45, legend = "none",
                font.xtickslab = c(10,"bold"), font.y = c(10,"bold") ) -> tmp
tmp + stat_compare_means()

lm(data = dat[dat$cell_type %in% c("Microglia","Astrocyte"),], formula = LOY_prop ~ cell_type + donor_organism.age) %>% summary()

```




```{r fig2c}
plots <- "h:/LOY/scLOY/scLOY_NOV18/plots/MARCH_12/MANUSCRIPT/MAIN/Fig2"

readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(l = l, min_cells = 75, sum_Y = 250) -> dat
factor(x = dat$cell_type, levels = c("Microglia","OPC","Oligodendrocyte","Astrocyte","Neuron"),
       labels = c("Microglia","OPC","OL","Astrocyte","Neuron")) -> dat$cell_type
paste0(dat$cohort, "_", dat$sample) -> dat$id


lapply(c("Microglia","OPC","OL","Astrocyte","Neuron"), function(x){
  
  dat[dat$CLEAN %in% c("Non-degenerative","AD") & dat$cell_type == x,] %>% 
  ggboxplot(x = "CLEAN", y = "LOY_prop", size = 0.3,
          add = "jitter", title = x, add.params = list(size = 0.8),
          fill = "CLEAN") -> p
  p + coord_cartesian(ylim = c(0,0.6)) -> p
  p %>% ggpar(palette = get_palette("npg",4)[c(1,4)], xlab = "", 
              ylab = "", legend = "none", x.text.angle = 45 ) -> p
  
return(p)
  
}) -> pl
ggarrange(plotlist = pl, ncol = 5, nrow = 1) -> p 

if(save == T){ 
ggsave(plot = p,
       filename = paste0(plots,"/LOY_BRAIN_CELL_TYPES_3000_1000_AD_v_CTRL.pdf"),
       dpi = "retina", units = "in", height = 3, width = 9)}


### Sample sizes
dat[dat$neuro_degen_diagnosis %in% c("Control","AD/MCI"),] %>% 
  dplyr::group_by(neuro_degen_diagnosis, cell_type) %>%
  dplyr::summarise(count = dplyr::n()) %>% 
  reshape2::dcast(cell_type ~ neuro_degen_diagnosis, value.var = "count")



### AD-LOY association

dat[dat$neuro_degen_diagnosis %in% c("Control","AD/MCI"),] %>% 
  dplyr::group_by(neuro_degen_diagnosis, cell_type) %>%
  dplyr::summarise(count = dplyr::n()) %>% 
  reshape2::dcast(cell_type ~ neuro_degen_diagnosis, value.var = "count")


```


```{r table_S1}


```

```{r table_S2}


```

```{r fig2_supp_age_dist}
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(l = l, min_cells = 75, sum_Y_ex = 250) -> dat
factor(x = dat$cell_type, levels = c("Microglia","OPC","Oligodendrocyte","Astrocyte","Neuron"),
       labels = c("Microglia","OPC","OL","Astrocyte","Neuron")) -> dat$cell_type
paste0(dat$cohort, "_", dat$sample) -> dat$id

dat[!is.na(dat$donor_organism.age),] -> dat




lapply(c("Microglia","OPC","OL","Astrocyte","Neuron"), function(x){
  
  if(x == "Microglia"){co <- 5}
  if(x == "OPC"){co <- 8}
  if(x == "OL"){co <- 7}
  if(x == "Astrocyte"){co <- 1}
  if(x == "Neuron"){co <- 6}
  
  dat[dat$cell_type==x,]$donor_organism.age %>% mean() %>% print()
   dat[dat$cell_type==x,]$donor_organism.age %>% length() %>% print()
  
  
  gghistogram(data = dat[dat$cell_type==x,], x = "donor_organism.age", fill= dittoSeq::dittoColors()[co] , 
              color= NA , title = x, alpha = 1,
            palette = dittoSeq::dittoColors()[co] ) %>% 
            ggpar(legend = "none", ylim = c(0,20), xlab = "", ylab = "", font.ytickslab = c(8),
                  font.xtickslab = c(8), x.text.angle = 45) -> p 
            
  
return(p)
  
}) -> pl

ggarrange(plotlist = pl, ncol = 5, nrow = 1) -> p 


if(save == T){ 
ggsave(plot = p,
       filename = paste0(plots,"/LOY_BRAIN_CELL_TYPES_3000_1000_SAMPLE_SIZE.pdf"),
       dpi = "retina", units = "in", height = 2.5, width = 8.5)}



```



```{r fig2_age_scatter}

#### AGE SCATTER #############
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(l, min_cells = 75, sum_Y_exp = 250 ) -> dat

cut(x = dat$donor_organism.age, breaks = c(-20,0,30,60,80,100),
    labels = c("Prenatal","0-29","30-59","60-79","80-100")) -> dat$quantile_age
dat[!is.na(dat$donor_organism.age),] -> tmp


factor(x = tmp$cell_type, levels = c("Microglia","Astrocyte","Neuron","Oligodendrocyte","OPC"),
       labels = c("Microglia","Astrocyte","Neuron","OL","OPC")) -> tmp$cell_type

tmp[!is.na(tmp$donor_organism.age),] -> tmp
tmp[tmp$cell_type %in% c("Microglia","Astrocyte","Neuron","OL","OPC") &
      !(tmp$quantile_age %in% c("Prenatal")),] -> tmp

ifelse(tmp$neuro_degen_diagnosis %in% c("AD/MCI","HD","PD","ALS","LBD/FTLD"),
       "Neurodegenerative disorder", "Control") -> tmp$neuro_degen_diagnosis_broad

ifelse(tmp$neuro_degen_diagnosis == "AD/MCI", "AD/MCI", ifelse(tmp$neuro_degen_diagnosis == "Control",
"Control", "Other")) -> tmp$condensed

factor(tmp$cell_type, levels = c("Microglia","OPC","OL","Astrocyte","Neuron")) -> tmp$cell_type
tmp %>%
  ggscatter(x = "donor_organism.age",y = "LOY_prop", color = "condensed", add = "reg.line", 
            size = 0.9, add.params = list(color="black",size = 0.9),
            palette = get_palette(palette = "npg", 4)[c(1,3,4)]) + stat_cor() + facet_grid(~cell_type) -> p
            ggpar(p, font.xtickslab = c(10), x.text.angle = 45, xlab = "", ylab = "", legend = "bottom", 
                  legend.title = "Donor diagnosis") -> p

if(save == T){             
ggsave(plot = p,
       filename = paste0(plots,"/LOY_BRAIN_CELL_TYPES_3000_1000_AGE_SCATTER.pdf"),
       dpi = "retina", units = "in", height = 3.5, width = 8.5)}
            
            
```



```{r total_LOY_boxplots}

readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

#data.table::fread(paste0(output,"/LOY_prop_table.txt")) -> l
l[l$TOTAL_CELLS > 50 & l$sum_Y_exp > 200, ] -> dat
dat[!grepl(dat$file, pattern = "MD5893"),] -> dat
dat[!grepl(dat$cohort, pattern = "GSE131907_LUNG_NSCLC_cleaned_filtered"),] -> dat
dat[which(dat$sample=="Microglia_MO_MCI3"),]$neuro_degen_diagnosis <- "AD/MCI"
dat[which(dat$neuro_degen_diagnosis=="AD"),]$neuro_degen_diagnosis <- "AD/MCI"

cut(x = dat$donor_organism.age, breaks = c(-20,0,30,60,80,100),
    labels = c("Prenatal","0-29","30-59","60-79","80-100")) -> dat$quantile_age
dat[!is.na(dat$donor_organism.age),] -> tmp


factor(x = tmp$cell_type, levels = c("Microglia","Astrocyte","Neuron","Oligodendrocyte","OPC"),
       labels = c("Microglia","Astrocyte","Neuron","OL","OPC")) -> tmp$cell_type
factor(tmp$cell_type, levels = c("Microglia","OPC","OL","Astrocyte","Neuron")) -> tmp$cell_type
tmp[!is.na(tmp$donor_organism.age),] -> tmp
tmp[tmp$cell_type %in% c("Microglia","Astrocyte","Neuron","OL","OPC") &
      !(tmp$quantile_age %in% c("Prenatal")),] -> tmp

ifelse(tmp$neuro_degen_diagnosis %in% c("AD/MCI","HD","PD","ALS","LBD/FTLD"),
       "Neurodegenerative disorder", "Control") -> tmp$neuro_degen_diagnosis_broad
paste0(tmp$cohort, "_", tmp$sample) -> tmp$id

ifelse(tmp$neuro_degen_diagnosis == "AD/MCI", "AD/MCI", ifelse(tmp$neuro_degen_diagnosis == "Control",
"Control", "Other")) -> tmp$condensed


tmp %>% dplyr::group_by(id) %>% 
  dplyr::summarise(LOY = sum(LOY_cells), NORMAL = sum(NORMAL_cells), quantile_age, donor_organism.age) %>% 
  dplyr::mutate(LOY_prop. = LOY / (LOY + NORMAL)) %>%
  ggboxplot(y = "LOY_prop.", x = "quantile_age")


tmp %>% dplyr::group_by(id) %>% 
  dplyr::summarise(LOY = sum(LOY_cells), NORMAL = sum(NORMAL_cells), quantile_age, donor_organism.age, neuro_degen_diagnosis) %>% 
  dplyr::mutate(LOY_prop. = LOY / (LOY + NORMAL)) %>%
  ggscatter(y = "LOY_prop.", x = "donor_organism.age", add.params = list(color = "black"), palette = "npg",
            add = "reg.line", color = "neuro_degen_diagnosis") + stat_cor()


tmp %>%
  ggscatter(x = "donor_organism.age",y = "LOY_prop", color = "condensed", add = "reg.line", 
            size = 0.6, add.params = list(color="black",size = 0.7),
            palette = get_palette(palette = "npg", 4)[c(1,3,4)]) + stat_cor() + facet_grid(~cell_type) -> p
            ggpar(p, font.xtickslab = c(10), x.text.angle = 45, xlab = "", ylab = "", legend = "bottom", 
                  legend.title = "Donor diagnosis") -> p

if(save == T){             
ggsave(plot = p,
       filename = paste0(plots,"/LOY_BRAIN_CELL_TYPES_3000_1000_AGE_SCATTER.pdf"),
       dpi = "retina", units = "in", height = 3.5, width = 8.5)}
            
            
```














### other analyses
# Reviewer comment 3.6

```{r PAR_LOY_v_NORMAL}

readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3000_1000.txt") %>%
   as.data.table() -> l

filter_function(l = l, min_cells = 75, sum_Y = 250, cell_type = l$cell_type %>% unique()) -> dat

dat[dat$cell_type == "Monoycte",]$cell_type <- "Monocyte"

paste0(dat$sample, "_", dat$cohort) -> dat$ID


dat %>% dplyr::mutate(diff_PAR = log1p((expr_PAR_LOY - expr_PAR) / expr_PAR)) -> dat

dat %>% dplyr::filter(LOY_cells >= 20) -> dat
dat[dat$expr_PAR_LOY > 0, ] -> dat
dat[dat$expr_PAR > 1.5, ] -> dat

ggscatter(dat, x = "diff_PAR", y = "LOY_prop", color = "cell_type") %>%
  ggpar(x.text.angle = 45, font.xtickslab = c(10,"bold"), ylab = "LOY proportion", xlab = "") %>% ggpar(legend = "none") -> p

ggstripchart(dat, x = "cell_type", y = "diff_PAR", add = "mean_sd", add.params = list(color = "red"), 
             size = "LOY_cells", shape = 21, fill = "blue", color = "black", stroke = 0.8) %>% 
  ggpar(ylab = "logFC PAR (NORMAL v LOY)", xlab = "Cell type", font.xtickslab = c("bold",8),
        x.text.angle = 45) + geom_hline(yintercept = 0) -> p 


if(save == T){ 
ggsave(plot = p, filename = "a:/Dropbox/LOY_microglia_paper/GR_RESUB/MANUSCRIPT/PLOTS_FOR_REVIEWERS/plots/PAR_logFC_stripchart.pdf",
       dpi = "retina", width = 10, height = 6, units = "in"
        
      )
}


dat$sample %>% unique %>% length
dat$cell_type %>% unique %>% length




###################################### 1500 ######
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_1500_1000.txt") %>%
   as.data.table() -> l

filter_function(l = l, min_cells = 75, sum_Y = 250, cell_type = l$cell_type %>% unique()) -> dat

dat[dat$cell_type == "Monoycte",]$cell_type <- "Monocyte"

paste0(dat$sample, "_", dat$cohort) -> dat$ID


dat %>% dplyr::mutate(diff_PAR = log1p((expr_PAR_LOY - expr_PAR) / expr_PAR)) -> dat

dat %>% dplyr::filter(LOY_cells >= 20) -> dat
dat[dat$expr_PAR_LOY > 0, ] -> dat
dat[dat$expr_PAR > 1.5, ] -> dat

ggscatter(dat, x = "diff_PAR", y = "LOY_prop", color = "cell_type") %>%
  ggpar(x.text.angle = 45, font.xtickslab = c(10,"bold"), ylab = "LOY proportion", xlab = "") %>% ggpar(legend = "none") -> p

ggstripchart(dat, x = "cell_type", y = "diff_PAR", add = "mean_sd", add.params = list(color = "red"), 
             size = "LOY_cells", shape = 21, fill = "blue", color = "black", stroke = 0.8) %>% 
  ggpar(ylab = "logFC PAR (NORMAL v LOY)", xlab = "Cell type", font.xtickslab = c("bold",8),
        x.text.angle = 45) + geom_hline(yintercept = 0) -> p 

if(save == T){ 
ggsave(plot = p, filename = "a:/Dropbox/LOY_microglia_paper/GR_RESUB/MANUSCRIPT/PLOTS_FOR_REVIEWERS/plots/PAR_logFC_stripchart_1500.pdf",
       dpi = "retina", width = 10, height = 6, units = "in"
        
      )
}







readRDS(file = "e:/LOY/scLOY/processed_seurat/BRAIN/GSE160936_AD_GLIAL_introns_included.RDS" ) -> o 

## What is the PAR score for each cell type

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Neuron") -> o. 
ggviolin(o.@meta.data, x = "cell_ident_MV", y = "PAR_score1", 
         fill = "LOY", add = c("mean", "mean_sd")) %>% 
  ggpar(legend.title = "LOY", legend = "bottom", ylab = "PAR gene module score", palette = get_palette("aaas",2 )[c(2,1)],
        xlab = "Cell type", x.text.angle = 45, font.xtickslab = c("bold",9)) 
        

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & specific_tissue == "EC" ) -> o. 
ggviolin(o.@meta.data, x = "cell_ident_MV", y = "normPAR", fill = "LOY", add = c("mean", "mean_sd")) %>% 
  ggpar(legend.title = "LOY", ylab = "normPAR (log2)", palette = get_palette("aaas",2 )[c(2,1)],
        xlab = "Cell type", x.text.angle = 45, font.xtickslab = c("bold",9))

ggviolin(o.@meta.data, x = "cell_ident_MV", y = "PAR_score1", fill = "LOY", add = c("mean", "mean_sd")) %>% 
  ggpar(legend.title = "LOY", ylab = "PAR gene module score", palette = get_palette("aaas",2 )[c(2,1)],
        xlab = "Cell type", x.text.angle = 45, font.xtickslab = c("bold",9)) -> p


if(save == T){ 
ggsave(plot = p, 
       filename = "a:/Dropbox/LOY_microglia_paper/GR_RESUB/MANUSCRIPT/PLOTS_FOR_REVIEWERS/plots/PAR_score_violin_SMITH_AM.pdf",
       dpi = "retina", width = 10, height = 6, units = "in"
        
      )}




o@meta.data$`ptau immunostaining (%ptau positive cells):ch1` %>% as.numeric() -> o@meta.data$ptau
o@meta.data$`braak tangle stage:ch1` %>% as.factor() -> o@meta.data$braak
o@meta.data$`amyloid immunostaining (%area stained):ch1` %>% as.numeric() -> o@meta.data$amy

# 


lapply(X = o@meta.data$cell_ident_MV %>% unique(), FUN =function(x){ 
  
  subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == x) -> o. 
  factor(o.@meta.data$LOY, levels = c("NORMAL","LOY")) -> o.@meta.data$LOY_factor
  ifelse(o.@meta.data$LOY == "LOY", 0, 1) %>% as.numeric() -> o.@meta.data$LOY_dummy
  wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)
  lm(data = o.@meta.data, formula = LOY_dummy ~ PAR_score1 + nCount_RNA + nFeature_RNA + percent.mt + percent.rb) %>% summary() -> p
  p$coefficients %>% unlist() %>% as.data.frame()  -> df
  df$cell_type <- x
  rownames(df) -> df$factor
  return(df) 

}) -> l
data.table::rbindlist(l) -> l 


-log10(l$`Pr(>|t|)`) -> l$logP

reshape2::dcast(data = l, formula = cell_type ~ factor, value.var = "Pr(>|t|)") -> a
reshape2::dcast(data = l, formula = cell_type ~ factor, value.var = "t value") -> b
reshape2::dcast(data = l, formula = cell_type ~ factor, value.var = "logP") -> c

c[,-2] -> c 














lapply(X = o@meta.data$cell_ident_MV %>% unique(), FUN =function(x){ 
  
  subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == x) -> o. 
  factor(o.@meta.data$LOY, levels = c("NORMAL","LOY")) -> o.@meta.data$LOY_factor
  ifelse(o.@meta.data$LOY == "LOY", 0, 1) %>% as.numeric() -> o.@meta.data$LOY_dummy
  wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)
  lm(data = o.@meta.data, formula = PAR_score1 ~ LOY_factor + nCount_RNA + nFeature_RNA + percent.mt + percent.rb) %>% summary() -> p
  p$coefficients %>% unlist() %>% as.data.frame()  -> df
  df$cell_type <- x
  rownames(df) -> df$factor
  return(df) 

}) -> l
data.table::rbindlist(l) -> l 

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Astro") -> o. 
factor(o.@meta.data$LOY, levels = c("NORMAL","LOY")) -> o.@meta.data$LOY_factor
ggscatter(data = o.@meta.data, y = "PAR_score1", x = "LOY_factor") 
ggscatter(data = o.@meta.data, y = "PAR_score1", x = "nCount_RNA", add = "reg.line", facet.by = "LOY") + stat_cor() 

-log10(l$`Pr(>|t|)`) -> l$logP

reshape2::dcast(data = l, formula = cell_type ~ factor, value.var = "Pr(>|t|)") -> a
reshape2::dcast(data = l, formula = cell_type ~ factor, value.var = "t value") -> b
reshape2::dcast(data = l, formula = cell_type ~ factor, value.var = "logP") -> c

c[,-2] -> c 


pheatmap::pheatmap(mat = as.matrix(b %>% tibble::column_to_rownames(var = "cell_type")), display_numbers = T)


as.matrix(c %>% tibble::column_to_rownames(var = "cell_type")) -> mat
colnames(mat)[1] <- "LOY"
ifelse(test = mat >= 4, yes = "***", 
                        no = ifelse(mat >= 3, yes =  "**", no = 
                               ifelse(test = mat >= 2, 
                                      yes = "*", no = "ns"))) -> sig

as.matrix(c %>% tibble::column_to_rownames(var = "cell_type")) -> mat
names(mat) <- "LOY"
          
breaksList = seq(0, 20, by = 0.01)
pheatmap::pheatmap(mat = mat, 
                   display_numbers = sig, breaks = breaksList , 
                   color = colorRampPalette(brewer.pal(n = 8, name = "Reds"))(length(breaksList)))

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Astro" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Pericyte" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Brain endothelial" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Oligo" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Neuron" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "CD4+ T" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "OPC" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Erythroid" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)





readRDS(file = "e:/LOY/scLOY/processed_seurat/MERGED_COHORT/GSE178265_cleaned_filtered.RDS" ) -> o
subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" ) -> o.

ggviolin(o.@meta.data, x = "cell_ident_MV", y = "PAR_score1", fill = "LOY", add = c("mean", "mean_sd")) %>% 
  ggpar(legend.title = "LOY", ylab = "PAR gene module score", palette = get_palette("aaas",2 )[c(2,1)],
        xlab = "Cell type", x.text.angle = 45, font.xtickslab = c("bold",9)) -> p 


if(save == T){ 
ggsave(plot = p, 
       filename = "a:/Dropbox/LOY_microglia_paper/GR_RESUB/MANUSCRIPT/PLOTS_FOR_REVIEWERS/plots/PAR_score_violin_GSE178265.pdf",
       dpi = "retina", width = 10, height = 6, units = "in"
        
      ) }

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Microglia" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)
lm(data = o.@meta.data, formula = PAR_score1 ~ LOY + nCount_RNA + nFeature_RNA ) %>% summary()
ifelse(o.@meta.data$LOY == "LOY", 1, 0) %>% as.numeric() -> o.@meta.data$LOY_factor
lm(data = o.@meta.data, formula = LOY_factor ~ PAR_score1 + nCount_RNA + nFeature_RNA + percent.mt + percent.rb ) %>% summary() -> s

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Astro" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)
lm(data = o.@meta.data, formula = PAR_score1 ~ LOY + nCount_RNA + nFeature_RNA ) %>% summary()
ifelse(o.@meta.data$LOY == "LOY", 1, 0) %>% as.numeric() -> o.@meta.data$LOY_factor
lm(data = o.@meta.data, formula = LOY_factor ~ PAR_score1 + nCount_RNA + nFeature_RNA + percent.mt + percent.rb ) %>% summary() -> s

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Neuron" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)
lm(data = o.@meta.data, formula = PAR_score1 ~ LOY + nCount_RNA + nFeature_RNA )  %>% summary()
ifelse(o.@meta.data$LOY == "LOY", 1, 0) %>% as.numeric() -> o.@meta.data$LOY_factor
lm(data = o.@meta.data, formula = LOY_factor ~ PAR_score1 + nCount_RNA + nFeature_RNA + percent.mt + percent.rb ) %>% summary() -> s
s$coefficients

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "OPC" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)
lm(data = o.@meta.data, formula = PAR_score1 ~ LOY + nCount_RNA + nFeature_RNA ) %>% summary() 
ifelse(o.@meta.data$LOY == "LOY", 1, 0) %>% as.numeric() -> o.@meta.data$LOY_factor
lm(data = o.@meta.data, formula = LOY_factor ~ PAR_score1 + nCount_RNA + nFeature_RNA + percent.mt + percent.rb ) %>% summary() -> s
s$coefficients

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Pericyte" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)
lm(data = o.@meta.data, formula = PAR_score1 ~ LOY + nCount_RNA + nFeature_RNA ) %>% summary() 
ifelse(o.@meta.data$LOY == "LOY", 1, 0) %>% as.numeric() -> o.@meta.data$LOY_factor
lm(data = o.@meta.data, formula = LOY_factor ~ PAR_score1 + nCount_RNA + nFeature_RNA + percent.mt + percent.rb ) %>% summary() -> s
s$coefficients

subset(o, nCount_RNA > 3000 & nFeature_RNA > 1000 & donor_organism.sex == "male" & cell_ident_MV == "Brain endothelial" ) -> o. 
wilcox.test(x = o.@meta.data[o.@meta.data$LOY == "LOY",]$normPAR, y = o.@meta.data[o.@meta.data$LOY == "NORMAL",]$normPAR)
lm(data = o.@meta.data, formula = PAR_score1 ~ LOY + nCount_RNA + nFeature_RNA ) %>% summary() 
ifelse(o.@meta.data$LOY == "LOY", 1, 0) %>% as.numeric() -> o.@meta.data$LOY_factor
lm(data = o.@meta.data, formula = LOY_factor ~ PAR_score1 + nCount_RNA + nFeature_RNA + percent.mt + percent.rb ) %>% summary() -> s
s$coefficients

```




##### OTHER analyses
```{r effect_of_QC_on_LOY_prop}
readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3500_1000.txt") %>%
   as.data.table() -> l
filter_function(l = l, min_cells = 75, sum_Y = 0) -> dat


dat[dat$cell_type %in% c("Oligodendrocyte","Microglia","Astrocyte","Neuron","OPC","Pericyte"),] -> dat

paste0(dat$sample, "_", dat$cohort) -> dat$ID


dat %>% dplyr::mutate(diff_PAR = log1p((expr_PAR_LOY - expr_PAR) / expr_PAR)) -> dat


# lm(data = dat[dat$cell_type], formula = LOY_prop ~ expr_Y + sum_Y_exp) %>% summary()
ggscatter(data = dat, x = "expr_Y", y = "LOY_prop", color = "diff_PAR", 
          facet.by = "cell_type", size = 1.5) + geom_smooth(se = F, method = "lm") + 
          scale_color_viridis_c(direction = -1, limits = c(-0.4,0.4), oob = scales::squish) + 
          coord_cartesian(ylim = c(0,0.6))


ggscatter(data = dat, x = "sum_Y_exp", y = "LOY_prop", fill = "diff_PAR", shape = 21, stroke = 1,  
          facet.by = "cell_type", size = 2) + geom_smooth(se = F, method = "lm") + 
          scale_fill_viridis_c(direction = -1, limits = c(-0.4,0.4), oob = scales::squish, option = "B") + 
          coord_cartesian(ylim = c(0,0.6)) 

ggscatter(data = dat, x = "sum_Y_exp", y = "LOY_prop", fill = "diff_PAR", shape = 21, stroke = 1,  
          facet.by = "cell_type", size = "LOY_cells") + geom_smooth(se = F, method = "lm") + 
          scale_fill_viridis_c(direction = -1, limits = c(-0.4,0.4), oob = scales::squish, option = "B", end = 0.9) + 
          coord_cartesian(ylim = c(0,0.6)) + geom_vline(xintercept = 250, color = "grey")




readr::read_csv(
  "https://raw.githubusercontent.com/michaelcvermeulen/microglia-loss-of-y/main/data/LOY_tables/LOY_prop_table_3500_1000.txt") %>%
   as.data.table() -> l

filter_function(l = l, min_cells = 75, sum_Y = 250) -> dat
dat[dat$cell_type %in% c("Oligodendrocyte","Microglia","Astrocyte","Neuron","OPC","Pericyte"),] -> dat
paste0(dat$sample, "_", dat$cohort) -> dat$ID


dat %>% dplyr::mutate(diff_PAR = log1p((expr_PAR_LOY - expr_PAR) / expr_PAR)) -> dat
ggscatter(data = dat[dat$donor_organism.age >= 0,], 
          x = "donor_organism.age", y = "LOY_prop", fill = "diff_PAR", shape = 21, stroke = 1,  
          facet.by = "cell_type", size = "LOY_cells") + geom_smooth(se = F, method = "lm") + 
          scale_fill_viridis_c(direction = -1, limits = c(-0.4,0.4), oob = scales::squish, option = "B", end = 0.9) + 
          coord_cartesian(ylim = c(0,0.6)) 



dat %>% dplyr::filter(LOY_cells >= 20) -> dat
dat[dat$expr_PAR_LOY > 0, ] -> dat
dat[dat$expr_PAR > 1.5, ] -> dat


```